content/blog/migrating-vsphere-virtual-volumes-to-kubevirt-with-hpe-csi-driver-for-kubernetes.md

```diff
@@ -0,0 +1,432 @@
+
+In the first iteration, the vSphere vVol needs to be cloned into a PersistentVolume (PV) using a Persistent Volume Claim (PVC). The PVC is then attached to the KubeVirt VM as a disk.
+
+PVC creation may rely on pre-provisioned “static” PVs but more commonly a StorageClass is used to dynamically provision the PV. To allow the PV to clone a pree-xisting volume on the storage array the HPE CSI Driver for Kubernetes needs to be supplied a parameter with the corresponding volume name.
+
+The CSI driver supports a construct that allows Kubernetes users to supply their own values to StorageClass parameters called “allowOverrides”. In this exercise we need to craft a StorageClass that allows users to influence the “importVolumeName” (used for migration) and “importVolAsClone” (used for cloning).
+
+```yaml
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: legacy-migration
+provisioner: csi.hpe.com
+parameters:
+  csi.storage.k8s.io/controller-expand-secret-name: hpe-backend
+  csi.storage.k8s.io/controller-expand-secret-namespace: hpe-storage
+  csi.storage.k8s.io/controller-publish-secret-name: hpe-backend
+  csi.storage.k8s.io/controller-publish-secret-namespace: hpe-storage
+  csi.storage.k8s.io/node-publish-secret-name: hpe-backend
+  csi.storage.k8s.io/node-publish-secret-namespace: hpe-storage
+  csi.storage.k8s.io/node-stage-secret-name: hpe-backend
+  csi.storage.k8s.io/node-stage-secret-namespace: hpe-storage
+  csi.storage.k8s.io/provisioner-secret-name: hpe-backend
+  csi.storage.k8s.io/provisioner-secret-namespace: hpe-storage
+  csi.storage.k8s.io/fstype: ext4
+  description: "Volume cloned or imported by the HPE CSI Driver for Kubernetes"
+  allowOverrides: importVolumeName,importVolAsClone
+  compression: "false"
+  provisioningType: ""
+reclaimPolicy: Delete
+allowVolumeExpansion: true
+```
+
+Users then call out the parameter and the value they desire in the PVC using an annotation.
+
+```yaml
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: testdb-cloned
+  namespace: hpe-vmi
+  annotations:
+    csi.hpe.com/importVolAsClone: dat-mariadbl-56480669
+spec:
+  volumeMode: Block
+  storageClassName: legacy-migration
+  accessModes:
+    - ReadWriteMany
+  resources:
+    requests:
+      storage: 128Gi
+```
+
+So, where does “dat-mariadbl-56480669” come from? It’s the storage array volume name. It’s available to storage administrators of the storage array and it’s also available through the HPE Storage Integration Pack for VMware vCenter (SIP4VC) when used with HPE Alletra Storage MP B10000 and earlier platforms.
+
+![](/img/sip4vc.png "Virtual machine view in SIP4VC")
+
+Create the PVC and attach the PVC as a disk to the running KubeVirt VM.
+
+```shell
+$ kubectl create -f pvc-clone.yaml
+```
+
+Next, login to OKD.
+
+![](/img/attach-disk.png "Attach disk to VM")
+
+Once attached, a “migrate” playbook will mount the disk in inside the VM and fire up the database and test it.
```

Comment:
I think there's a typo in this sentence? "...playbook will mount the disk **in inside** the VM..." (you probably don't need that first "in")